Client/src/Components/DataCell/DataCell.js

```diff
@@ -1,47 +1,89 @@
-import Box from '@material-ui/core/Box'
-import Button from '@material-ui/core/Button'
-import { Container } from '@material-ui/core'
-import React from 'react'
-
-/**
- * This component handles receiving the json locally then sending the file for processing
- * to the correct API.
- */
-export default function DataCell() {
-  /**
-     * Upon submission, the JSON file is extracted from the event and must be appended to formData
-     * to be sent with API request.
-     */
-  const handleSubmit = async (e) => {
-    e.preventDefault()
-
-    const json = e.target.file.files[0]
-
-    const formData = new FormData()
-    formData.append('file', json)
-
-    const options = {
-      method: 'POST',
-      body: formData,
-    }
-    await fetch('http://localhost:4000/dataupload', options)
-      .then(resp => resp.json())
-      .then(result => {
-        console.log(result[0])
-      })
-  }
-
-  return (
-    <Container>
-      <Box border={30} p={4} borderColor="primary">
-        <form onSubmit={handleSubmit}>
-          <img src={require('./uploadimage.png')} alt="Visual of clouds"></img>
-          <div>
-            <input type="file" id="file" accept=".json" />
-            <Button type="submit" variant="contained"> Upload this file! </Button>
-          </div>
-        </form>
-      </Box>
-    </Container>
-  )
-}
+//Verify that material-ui lab and material-ui core are installed before running this file!
+// npm install @material-ui/lab
+
+import Box from '@material-ui/core/Box'
+import Button from '@material-ui/core/Button'
+import { Container } from '@material-ui/core'
+import { Snackbar } from '@material-ui/core'
+import React, { useState } from 'react'
+import { Alert } from '@material-ui/lab'
+
+
+/**
+ * This component handles receiving the json locally then sending the file for processing
+ * to the correct API.
+ */
+export default function DataCell() {
+    const [open, setOpen] = useState(false);
+
+    /**
+     * Upon submission, the JSON file is extracted from the event and must be appended to formData
+     * to be sent with API request.
+     */
+  const handleSubmit = async (e) => {
+    e.preventDefault()
+
+        const json = e.target.jsonFile.files[0];
+        try {
+            //so if this here is not proper json extension it will catch
+            var ext = getExtension(json.name);
+            if (ext != '.json') {
+                throw "File extension is not supported";

```

Comment:
its up to you to throw, I dont mind, but exceptions are meant to cover something that is not in the developer's control, here however instead of throwing we could setOpen(true) and return

Reply:
i have changed the throw to a console.error() instead to avoid confusion for the time being.
just to clarify, what should we try to return?

Reply:
for now, file extension check has been moved out of try/catch block